src/Microsoft.IdentityModel.Tokens/InternalValidators.cs

```diff
@@ -0,0 +1,144 @@
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//------------------------------------------------------------------------------
+
+using System;
+using System.Text;
+using Microsoft.IdentityModel.Logging;
+using TokenLogMessages = Microsoft.IdentityModel.Tokens.LogMessages;
+
+namespace Microsoft.IdentityModel.Tokens
+{
+    /// <summary>
+    /// Validators meant to be kept internal
+    /// </summary>
+    internal static class InternalValidators
+    {
+        /// <summary>
+        /// Called after signature validation has failed. Will always throw an exception.
+        /// </summary>
+        /// <exception cref="SecurityTokenSignatureKeyNotFoundException">
+        /// If the lifetime and issuer are valid
+        /// </exception>
+        /// <exception cref="SecurityTokenUnableToValidateException">
+        /// If the lifetime or issuer are invalid
+        /// </exception>
+        internal static void ValidateLifetimeAndIssuerAfterSignatureNotValidatedJwt(SecurityToken securityToken, DateTime? notBefore, DateTime? expires, string kid, TokenValidationParameters validationParameters, StringBuilder exceptionStrings)
+        {
+            bool validIssuer = false;
+            bool validLifetime = false;
+
+            try
+            {
+                Validators.ValidateLifetime(notBefore, expires, securityToken, validationParameters);
+                validLifetime = true;
+            }
+            catch (Exception)
+            {
+                // validLifetime will remain false
+            }
+
+            try
+            {
+                Validators.ValidateIssuer(securityToken.Issuer, securityToken, validationParameters);
+                validIssuer = true;
+            }
+            catch (Exception)
+            {
+                // validIssuer will remain false
+            }
+
+            if (validLifetime && validIssuer)
+                throw LogHelper.LogExceptionMessage(new SecurityTokenSignatureKeyNotFoundException(LogHelper.FormatInvariant(TokenLogMessages.IDX10501, kid, exceptionStrings, securityToken)));
+            else
```

Comment:
Because of the bitfield enum I don't think we can reduce it that way. it would become roughly

if (validLifetime && validIssuer) => Key not found
if (validLifetime && !validIssuer) => UnableToValidate(invalid issuer)
if (!validLifetime && validIssuer) => UnableToValidate(invalid lifetime)
else => UnableToValdiate(invalid issuer & invalid lifetime)

I think